### PR TITLE
[WFLY-11943] add code to retrieve resteasy endpoint data and insert i…

### PR DESCRIPTION
**Pull request by rsearls**
…nto model.

https://issues.jboss.org/browse/WFLY-11943

This PR is part of RFE https://issues.jboss.org/browse/EAP7-606
QA requested this be created for access during their Pre-Verification phase.
It should NOT be considered for merging at this time.

For new features ensure as well:
- [ ] Analysis was done 
     Document at https://github.com/wildfly/wildfly-proposals/pull/201
- [ ] Test Plan has been done
    Document at https://polarion.engineering.redhat.com/polarion/#/project/EAP7/wiki/RFEs_7_3/EAP7-606
- [ ] Tests were verified in advance
    This is in progress

### CODE REVIEW

**Comment by marekkopecky** [comment]
This PR is currently blocked by https://issues.jboss.org/browse/WFLY-12091

**Comment by bstansberry** [comment]
@rsearls https://github.com/wildfly/wildfly/commit/3960a17aefb73072d49816b83e6d2c41db37386d fixes your problem. Galleon provisions a server's config files by calculating and executing management ops and letting the server persist the xml. But JaxrsSubsystemAdd was not storing the new statistics-enabled attribute to the model so nothing got persisted.

Your PR should include an update to https://github.com/wildfly/wildfly/blob/master/jaxrs/src/test/java/org/jboss/as/jaxrs/JaxrsSubsystemTestCase.java#L40 so the test case covers this new attribute.

**Comment by rsearls** [comment]
@bstansberry 
Thanks I have made the needed changes.

**Comment by jfdenise** [comment]
@rsearls instead of duplicating the attribute in the galleon layer and basic profile would you mind define a jaxrs feature-group and reference the feature-group from the jaxrs layer and basic-profile? it will help maintenance. Thank-you.

**Comment by rsearls** [comment]
Brian,

Would you care to comment on Jean-François's request for me to change the
location of the addition of the jaxrs attribute.

**Comment by bstansberry** [comment]
@rsearls +1 to his suggestion.  I haven't even built this, so there may be typos etc but I think the last commit at https://github.com/bstansberry/wildfly/commits/testPullRequest12472 is what's needed.

**Comment by rsearls** [comment]
Ok thanks.
galleon-pack/src/main/resources/feature_groups/jaxrs.xml   has been added.

**Comment by rsearls** [comment]
Addition complete

**Comment by jfdenise** [comment]
Hi @rsearls , you added the feature-group, that is nice, but you are not making use of it in basic-profile and jaxrs layer. You should replace the jaxrs feature by  <feature-group name="jaxrs"/>

**Comment by rsearls** [comment]
@jfdenise 
Please point me to sample code I can follow.

**Comment by jfdenise** [comment]
You can look at brian repo: https://github.com/bstansberry/wildfly/commit/58364fbff3006fa4613f366cc9ff7f47211df482#diff-7c7e29588d041416fe03af41b27ffefeR105

**Comment by rsearls** [comment]
Thanks, adjustments have been made.

**Comment by jfdenise** [comment]
@rsearls , Thank-you.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/deploy/runtime/jaxrs/HelloApplication.java [hunk 1]
```diff
@@ -38,6 +38,8 @@
         final Set<Class<?>> classes = new HashSet<>();
         classes.add(HelloResource.class);
         classes.add(PureProxyEndPoint.class);
+        classes.add(HelloBadResource.class);
```
Comment (marekkopecky) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/deploy/runtime/jaxrs/HelloApplication.java]: This change causes regression of JaxrsRuntimeNameTestCase. JaxrsRuntimeNameTestCase uses HelloApplication.java as well, but HelloBadResource is not added to the deployment from JaxrsRuntimeNameTestCase. So JaxrsRuntimeNameTestCase fails.

Comment (marekkopecky) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/deploy/runtime/jaxrs/HelloApplication.java]: This change causes regression of JaxrsRuntimeNameTestCase. JaxrsRuntimeNameTestCase uses HelloApplication.java as well, but HelloBadResource is not added to the deployment from JaxrsRuntimeNameTestCase. So JaxrsRuntimeNameTestCase fails.

**Review by marekkopecky — CHANGES_REQUESTED** [review]

**Comment by marekkopecky** [comment]
AccessConstraintUtilizationTestCase#testConstraintUtilization is failing with:
```
[ERROR] Failures: 
[ERROR]   AccessConstraintUtilizationTestCase.testConstraintUtilization:154 {"outcome" => "failed","failure-description" => "WFLYCTL0216: Management resource '[
    (\"core-service\" => \"management\"),
    (\"access\" => \"authorization\"),
    (\"constraint\" => \"sensitivity-classification\"),
    (\"type\" => \"messaging-activemq\")
]' not found","rolled-back" => true}
[ERROR]   PermissionsCoverageTestCase.testTheEntireDomainTreeHasPermissionsDefined:70 {"outcome" => "failed","failure-description" => "WFLYCTL0158: Operation handler failed: java.util.MissingResourceException: Can't find resource for bundle java.util.PropertyResourceBundle, key jaxrs.deployment.rest-resource-paths.method_statistics","rolled-back" => true}
```

**Comment by rsearls** [comment]
@jfdenise  @bstansberry 
Having added a single attribute to the existing subsystem=jaxrs appears to
break wfly's ability to properly discover this subsystem.  The following 
integration tests fail.  All failures appear related to not properly finding 
the jaxrs subsystem definition.

Who can I consult to figure out how to resolve these issues?
 
Failing test:
rbac  org/jboss/as/test/integration/mgmt/access/PermissionsCoverageTestCase.java 
basic org/jboss/as/test/integration/jmx/ModelControllerMBeanTestCase.java 

-------
org/jboss/as/test/integration/mgmt/access/PermissionsCoverageTestCase.java

  Reports failure from AbstractOperationContext line 1037 msg

    "WFLYCTL0158: Operation handler failed: java.util.MissingResourceException: 
    Can't find resource for bundle java.util.PropertyResourceBundle, 
    key jaxrs.deployment.rest-resource-paths.method_statistics","rolled-back" => true}

------
basic org/jboss/as/test/integration/jmx/ModelControllerMBeanTestCase.java
    ResourceAccessControl getResourceAccessWithInstanceNotFoundExceptionIfNotAccessible(
              ObjectName name, PathAddress address, boolean operations) throws InstanceNotFoundException {
        ResourceAccessControl accessControl = getResourceAccess(address, operations);
        if (!accessControl.isAccessibleResource()) {
            throw JmxLogger.ROOT_LOGGER.mbeanNotFound(name);
        }
        return accessControl;
    }

  Variable accessControl reports a false address
        protect: true
        value: {"address" => false}
  
  Variables name and address are report to be
        name: jboss.as:subsystem=jaxrs
        address: [("subsystem" => "jaxrs")]

**Comment by jfdenise** [comment]
@rsearls , don'y you have to add something to: https://github.com/wildfly/wildfly/blob/master/jaxrs/src/main/resources/org/jboss/as/jaxrs/LocalDescriptions.properties file?

**Comment by rsearls** [comment]
Thanks that resolved the issue with PermissionsCoverageTestCase.

**Comment by jfdenise** [comment]
@rsearls , just a note, I can see a lot of formatting changes in your commit, that creates noise when reviewing. That would help to remove them.

**Comment by rsearls** [comment]
@jfdenise 
The failure with org/jboss/as/test/integration/jmx/ModelControllerMBeanTestCase.java still remains.
I need assistance with this one as well.

**Comment by rsearls** [comment]
Regression in ModelControllerMBeanTestCase.java has been addressed

**Comment by darranl** [comment]
@rsearls I am just trying to tidy up the WildFly PR queue as we have a lot of stale PRs in the queue, it looks like this now has a set of merge conflicts that will need to be addressed - if this is continuing could you also please confirm the state so we can kick off review again and see if we are ready for a merge.  Alternatively can we please close the PR for now and re-open once ready.

**Pull request closed without merging**